lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -210,24 +248,47 @@ def _user_requested_to_enable(self, plugin: OpenSearchPlugin) -> bool:
     def _is_enabled(self, plugin: OpenSearchPlugin) -> bool:
         """Returns true if plugin is enabled."""
         # If not requested to be disabled, check if options are configured or not
-        are_configs_enabled = plugin.config().config_entries_to_add
-        stored_plugin_conf = self._opensearch_config.get_plugin(are_configs_enabled)
-        for key, val in stored_plugin_conf.items():
-            if are_configs_enabled.get(key) != val:
+        try:
+            are_configs_enabled = plugin.config().config_entries_to_add
+            stored_plugin_conf = self._opensearch_config.get_plugin(are_configs_enabled)
+            # Compare dicts
+            list_a = list(are_configs_enabled.keys())
+            list_a.sort()
+            list_b = list(stored_plugin_conf.keys())
+            list_b.sort()
+            if list_a != list_b:
+                # Keys do not match
                 return False
-        return True
+            for key, val in stored_plugin_conf.items():
+                if are_configs_enabled.get(key) != val:
+                    return False
+            return True
+        except (KeyError, OpenSearchPluginError) as e:
+            logger.warning(f"_is_enabled: error with {e}")
+            return False
 
     def _needs_upgrade(self, plugin: OpenSearchPlugin) -> bool:
         """Returns true if plugin needs upgrade."""
-        current_version = plugin.version
-        num_points = len(self._opensearch.version.split("."))
-        return self._opensearch.version != current_version[:num_points]
+        try:
+            # Plugins may have versioning slightly different than opensearch's:
+            version = self._opensearch.version
+            num_points = min(len(plugin.version.split(".")), len(version.split(".")))
+            return version[:num_points] != plugin.version[:num_points]
+        except OpenSearchError:
+            # A race condition may happen if plugins with relation events happen too early
+            # then, this method will be called before the cluster is ready.
+            # This exception should trigger a Plugin-specific error, which will be used
```

Comment:
Could you explain why?

Reply:
I will remove this try/catch, it was likely a left over before my changes to check the cluster green/yellow health as above.